src/btree/bt_handle.c

```diff
@@ -1017,12 +1017,18 @@ int
 __wt_btree_switch_object(WT_SESSION_IMPL *session, uint32_t objectid)
 {
     WT_BM *bm;
+    WT_BTREE *btree;
+
+    btree = S2BT(session);
+    /* If the btree is readonly, there is nothing to do. */
+    if (F_ISSET(btree, WT_BTREE_READONLY))
+        return (0);
```

Comment:
I suggest adding a python test case for this.  `test_tiered17.py` has other readonly tests, so it's probably a good place for it.

Reply:
Done.